src/goto-instrument/nondet_static.cpp

```diff
@@ -136,3 +140,59 @@ void nondet_static(goto_modelt &goto_model)
   const namespacet ns(goto_model.symbol_table);
   nondet_static(ns, goto_model.goto_functions);
 }
+
+/// Second main entry point of the module. Nondeterministically initializes
+/// global scope variables, except for constants (such as string literals, final
+/// fields), internal variables (such as CPROVER and symex variables,
+/// language specific internal variables) and variables passed to except_value.
+/// \param [out] goto_model: Existing goto-model to be updated.
+/// \param except_values: list of symbol names that should not be updated.
+void nondet_static(
+  goto_modelt &goto_model,
+  const optionst::value_listt &except_values)
+{
+  const namespacet ns(goto_model.symbol_table);
+  std::set<std::string> to_exclude;
+
+  for(auto const &except : except_values)
+  {
+    const bool file_prefix_found = except.find(":") != std::string::npos;
+
+    if(file_prefix_found)
+    {
+      to_exclude.insert(except);
+      if(has_prefix(except, "./"))
+      {
+        to_exclude.insert(except.substr(2, except.length() - 2));
+      }
+      else
+      {
+        to_exclude.insert("./" + except);
+      }
+    }
+    else
+    {
+      irep_idt symbol_name(except);
+      symbolt lookup_results = ns.lookup(symbol_name);
+      to_exclude.insert(
+        id2string(lookup_results.location.get_file()) + ":" + except);
+    }
+  }
+
+  symbol_tablet &symbol_table = goto_model.symbol_table;
+
+  for(symbol_tablet::iteratort symbol_it = symbol_table.begin();
+      symbol_it != symbol_table.end();
+      symbol_it++)
```

Comment:
Range for if possible (otherwise ++symbol_it and you can use auto for the iterator type)